### PR TITLE
Remove `getPseudoRandomNumber` from PRNG precompiles

### LINKED ISSUE #3702: Remove `getPseudoRandomNumber` from PRNG precompiles
Based on the below reasons , remove `getPseudoRandomNumber` from PRNG precompiles :
1. Since it might be confusing to have `getPseudoRandomNumber` that generates different numbers each time
2. It is very expensive for users to generate seed and use the seed in this function to generate same pseudorandom number for same seed, instead users can directly do first 32 bits of seed % range in solidity easily.

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
Fixes #3702 

Since it might be confusing to have `getPseudoRandomNumber` that generates different numbers each time and it is very expensive for users to generate seed and use the seed in this function to generate same pseudorandom number for same seed, it was decided that users can directly do `first 32 bits of seed % range` in solidity easily. For the above reasons removing the function from precompile

### CODE REVIEW

**Review by shemnon — DISMISSED** [review]

**Review by Nana-EC — APPROVED** [review]
LG
Though would be useful to update PR comments with intended changes and context for it

File: test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/PrngPrecompileSuite.java [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+        final var lessThan4Bytes = "lessThan4Bytes";
+        return defaultHapiSpec("functionCallWithLessThanFourBytesFailsGracefully")
+                .given(
+                        cryptoCreate(bob),
+                        uploadInitCode(THE_GRACEFULLY_FAILING_PRNG_CONTRACT),
+                        contractCreate(THE_GRACEFULLY_FAILING_PRNG_CONTRACT))
+                .when(
+                        sourcing(
+                                () ->
+                                        contractCall(
```
Comment (shemnon) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/PrngPrecompileSuite.java:170]: The BDD code will transalte this into a 4 byte call.  we need the `withExplicitParams` method to about 3 bytes.

File: test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/PrngPrecompileSuite.java [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+    private HapiApiSpec invalidLargeInputFails() {
+        final var prng = THE_GRACEFULLY_FAILING_PRNG_CONTRACT;
+        final var largeInputCall = "largeInputCall";
+        return defaultHapiSpec("invalidLargeInputFails")
+                .given(cryptoCreate(bob), uploadInitCode(prng), contractCreate(prng))
+                .when(
+                        sourcing(
+                                () ->
+                                        contractCall(prng, "performLargeInputCall")
```
Comment (shemnon) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/PrngPrecompileSuite.java:108]: There is no extra data passed in. Perhaps `withExplicitParams` with a very large byte array?

File: test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/PrngPrecompileSuite.java [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+
+    private HapiApiSpec emptyInputCallFails() {
+        final var prng = THE_GRACEFULLY_FAILING_PRNG_CONTRACT;
+        final var emptyInputCall = "emptyInputCall";
+        return defaultHapiSpec("emptyInputCallFails")
+                .given(cryptoCreate(bob), uploadInitCode(prng), contractCreate(prng))
+                .when(
+                        sourcing(
+                                () ->
+                                        contractCall(prng, "performEmptyInputCall")
```
Comment (shemnon) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/PrngPrecompileSuite.java:78]: `withExplicitParams` with a zero length array is what I think is needed.

File: test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/PrngPrecompileSuite.java (hunk 2 above)
Comment (Neeharika-Sompalli) [reply test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/PrngPrecompileSuite.java]: Thank you ! Fixed to use `withExplicitParams`

File: test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/PrngPrecompileSuite.java (hunk 3 above)
Comment (Neeharika-Sompalli) [reply test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/PrngPrecompileSuite.java]: Fixed to use `withExplicitParams`

**Review by Nana-EC — APPROVED** [review]

**Review by tinker-michaelj — APPROVED** [review]
LGTM

**Review by shemnon — APPROVED** [review]

**Pull request merged**